### ISSUE TITLE
No /dev/shm device on compute nodes

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#149]
**Reported by jcsiadal on 8 Nov 2005 21:39 UTC**
```
The /dev/shm device is missing on compute nodes. We use 
this device for various tests/runs in the HPC environment.

This device is added to the fstab file by default in a normal 
installation. The oscarmaster script doesn't include this 
option when it builds the fstab on compute nodes.

Since this is standard configuration for Red Hat head node 
installs, and is sometimes needed for HPC applications, the 
oscarmaster script should inlcude the following line during 
the fstab creation:

none   /dev/shm   tmpfs   defaults   0 0

```

**Comment by dikim33** [comment]
**Comment by prg3 on 24 Nov 2005 05:06 UTC**
```

This looke like an issue with SiSuite that Oscar uses to
generate images.  Or rather, in trying to get this working,
we have found that sisuite does not have a good way to allow
for the mounting of /dev/shm.  Email bsing sent to
sisuite-devel right away to figure out resolution.

Leaving ticket open until resolved in sisuite.
```

**Comment by dikim33** [comment]
**Comment by bli on 26 Jun 2006 02:09 UTC**
The man page for ```mksidisk``` gave me hints on how to solve this:

```
             # There are defaults for /proc,/dev/pts &
             # /mnt/floppy,
             # but if you want to change them, just add lines
             # here.
             # The values here are the defaults.
             #/dev/fd0   -       auto   /mnt/floppy noauto,owner
             #/proc      -       proc   /proc   defaults
             #/dev/pts   -       devpts /dev/pts mode=622
```

I added the following to the ```.disk``` files and the resulting image/compute node has ```/dev/shm``` mounted:

```
/dev/shm         -    tmpfs /dev/shm   defaults
```

**Comment by dikim33** [comment]
**Comment by bli on 28 Jun 2006 06:51 UTC**
(In [5061]) Fixes #151.  This adds /dev/shm to all the compute nodes.
